public/app/features/alerting/unified/components/silences/SilencesFilter.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useState, FormEvent } from 'react';
+import { css } from '@emotion/css';
+import { Label, Icon, Input, Tooltip, RadioButtonGroup, useStyles2, Button } from '@grafana/ui';
+import { GrafanaTheme2, SelectableValue } from '@grafana/data';
+import { useQueryParams } from 'app/core/hooks/useQueryParams';
+import { getFiltersFromUrlParams } from '../../utils/misc';
+import { debounce } from 'lodash';
+import { SilenceState } from 'app/plugins/datasource/alertmanager/types';
+
+export const SilencesFilter = () => {
+  const [filterKey, setFilterKey] = useState<number>(Math.floor(Math.random() * 100));
+  const queryStringKey = `queryStringKey-${filterKey}`;
+  const [queryParams, setQueryParams] = useQueryParams();
+  const { queryString, silenceState } = getFiltersFromUrlParams(queryParams);
+  const styles = useStyles2(getStyles);
+
+  const handleQueryStringChange = debounce((e: FormEvent<HTMLInputElement>) => {
+    const target = e.target as HTMLInputElement;
+    setQueryParams({ queryString: target.value || null });
+  }, 600);
+
+  const handleSilenceStateChange = (state: string) => {
+    setQueryParams({ silenceState: state });
+  };
+
+  const clearFilters = () => {
+    setQueryParams({
+      queryString: null,
+      silenceState: null,
+    });
+    setTimeout(() => setFilterKey(filterKey + 1), 100);
+  };
+
+  const stateOptions: SelectableValue[] = Object.entries(SilenceState).map(([key, value]) => ({
+    label: key,
+    value,
+  }));
+
+  return (
+    <div className={styles.flexRow}>
+      <div className={styles.rowChild}>
+        <Label>
+          <Tooltip
+            content={
+              <div>
+                Filter silences by matchers using a comma separated list of matchers, ie:
+                <pre>{`severity=critical, instance=~cluster-us-.+`}</pre>
+              </div>
+            }
+          >
+            <Icon name="info-circle" />
+          </Tooltip>
+          Search by matchers
+        </Label>
+        <Input
+          key={queryStringKey}
```

Comment:
I've run into issues previously with the value not getting updated when clearing the filters but perhaps the nullish coalescing may resolve that. I'll look into it.

Reply:
I tried just using the value approach and it works if I also remove the `debounce` on the change handler. Seems like a reasonable tradeoff

Reply:
On second thought I think it's still good to keep the `queryStringKey` based on some observations. Without the debouncing of the event handler there can be some async race conditions that can occur between typing and the `queryString` getting set. So if the user types even moderately quickly it can cause characters to get overwritten. I tried to overcome that by debouncing the setting of the queryParams but it required me to create another piece of state to track the query input and just felt a little heavy handed.